From 7deaf0628a0884eedd176b6b58ed4aff5566e818 Mon Sep 17 00:00:00 2001
From: Ryan Liptak <squeek502@hotmail.com>
Date: Tue, 2 Sep 2025 02:26:01 -0700
Subject: [PATCH 086/133] zstd.Decompress: Assert buffer length requirements as
 early as possible
X-Developer-Signature: v=1; a=openpgp-sha256; l=868; i=xtex@astrafall.org;
 h=from:subject; bh=DyoLx8DZWYwU9n6MSdlf5bgYLCKsqzZd1OQYi16DZyo=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78thVN2yp3pu37s6K2TbmTLxx0hOnBW3O5Ofd+2wO2
 wf9exJzOyayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmWvULw2/2RbueKU5+UFVnMpFXatJZ1dSqRxIzWVMt+6YWTHIRPX2U4Z9Fe22
 vzqRI7vIZHxd8WvuPQSe2oMnu5tabJscfz/aSNOAHAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Without this assert, providing a buffer that's smaller than required results in more cryptic assertion failures later on.

(cherry picked from commit f872dd03da64e5b53d3a5c7a95344b5b64a7aacb)
---
 lib/std/compress/zstd/Decompress.zig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/std/compress/zstd/Decompress.zig b/lib/std/compress/zstd/Decompress.zig
index 0fc807333fc9..09df699b1f0f 100644
--- a/lib/std/compress/zstd/Decompress.zig
+++ b/lib/std/compress/zstd/Decompress.zig
@@ -92,6 +92,7 @@ const indirect_vtable: Reader.VTable = .{
 ///
 /// Otherwise, `buffer` has those requirements.
 pub fn init(input: *Reader, buffer: []u8, options: Options) Decompress {
+    if (buffer.len != 0) assert(buffer.len >= options.window_len + zstd.block_size_max);
     return .{
         .input = input,
         .state = .new_frame,
-- 
2.52.0

